src/features/boundary_selector/atoms/boundaryMarkerAtom.ts

```diff
@@ -74,51 +84,58 @@ export const boundaryMarkerAtom = createAtom(
       fc: GeoJSON.FeatureCollection,
       boundaryId: string,
     ) => {
-      const feature = fc.features.find((boundary) => boundary.id === boundaryId)!;
-      return highlightedGeometry.set(feature);
+      const feature = fc.features.find((boundary) => boundary.id === boundaryId);
+      return highlightedGeometry.set(
+        feature || { type: 'FeatureCollection', features: [] },
+      );
     };
 
     // Marker creation
     if (state.isEnabled) {
       onChange('boundaryResourceAtom', (resource) => {
-        const { data: featureCollection } = resource;
+        const { data: featureCollection, loading } = resource;
+
         const coordinates = get('clickCoordinatesAtom');
         const map = get('currentMapAtom');
         if (!coordinates || !map) return;
 
         schedule((dispatch) => {
-          const options = resource.data
-            ? constructOptionsFromBoundaries(resource.data)
-            : LOADING_OPTIONS;
+          let selectOptions =
+            resource.data && constructOptionsFromBoundaries(resource.data);
+          if (!selectOptions?.length)
+            selectOptions = loading ? LOADING_OPTION : NO_DATA_OPTION;
 
           const markerData = getSelectorWithOptions(
-            options,
+            selectOptions,
 
             // onOptionSelect:
             (boundaryId) => {
               if (!featureCollection) return;
               const selectedFeature = featureCollection.features.find(
                 (f) => f.id === boundaryId,
-              )!;
+              );
 
               const actions: Action[] = [
-                updateFocusedGeometryAction(selectedFeature),
                 toolbarControlsAtom.disable('BoundarySelector'),
                 updateBoundaryLayerAction(
                   { type: 'FeatureCollection', features: [] },
                   boundaryId,
                 ),
               ];
 
+              if (!selectedFeature) return dispatch(actions);
```

Comment:
Why you return dispatch from schedule?

Reply:
okay!
return was used just to end function